packagestate/packages_state_manager.go

```diff
@@ -49,16 +59,16 @@ type packageStates struct {
 	PackageStates   map[string]*packageState `json:"package_states"`
 }
 
-// NewPackagesStateManager creates a new PackagesStateManager
-func NewPackagesStateManager(logger *zap.Logger, jsonPath string) *PackagesStateManager {
-	return &PackagesStateManager{
+// NewFilePackagesStateManager creates a new PackagesStateManager
+func NewFilePackagesStateManager(logger *zap.Logger, jsonPath string) PackageStateManager {
```

Comment:
You mean the below syntax?

```go
var Thing Interface = (*Implementation)(nil)
```

That's not required in Go. Sometimes it's useful if you don't have a "constructor" method that returns the interface or the interface you're implementing is an another package. It's a personal preference. I felt like it wasn't needed here because of the constructor.